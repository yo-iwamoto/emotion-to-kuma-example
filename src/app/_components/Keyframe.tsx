'use client';

import { css, keyframes } from '@emotion/react';

export function Keyframe() {
  return <p css={styles.text}>Keyframe</p>;
}

const styles = {
  text: css`
    color: purple;
    font-size: 40px;
    animation: ${keyframes`
      0% {
        opacity: 0;
      }
      100% {
        opacity: 1;
      }
    `} 1s linear;
  `,
};
