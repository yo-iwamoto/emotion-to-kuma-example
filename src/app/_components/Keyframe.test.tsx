import { Keyframe } from './Keyframe';
import { render } from '@testing-library/react';

describe('Keyframe', () => {
  it('renders unchanged', () => {
    const { asFragment } = render(<Keyframe />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('has color style', () => {
    const { getByText } = render(<Keyframe />);

    expect(getByText('Keyframe')).toHaveStyleRule('color', 'purple');
  });
});
