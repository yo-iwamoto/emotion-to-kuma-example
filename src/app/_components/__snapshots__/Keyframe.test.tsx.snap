// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Keyframe > renders unchanged 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  0% {
    opacity: 0;
  }

  100% {
    opacity: 1;
  }
}

.emotion-0 {
  color: purple;
  font-size: 40px;
  -webkit-animation: animation-0 1s linear;
  animation: animation-0 1s linear;
}

<p
    class="emotion-0"
  >
    Keyframe
  </p>
</DocumentFragment>
`;
